iotilecore/iotile/core/hw/transport/adapter/mixin_notifications.py

```diff
@@ -190,8 +196,35 @@ def remove_monitor(self, handle):
             self._deferred_adjustments = []
             self._currently_notifying = False
 
-    def fire_event(self, conn_string, name, event):
-        """Fire an event without waiting.
+    def notify_event(self, conn_string, name, event):
+        """Notify an event.
+
+        This will move the notification to the background event loop and
+        return immediately.  It is useful for situations where you cannot
+        await notify_event but keep in mind that it prevents back-pressure
+        when you are notifying too fast so should be used sparingly.
+
+        Note that calling this method will push the notification to a
+        background task so it can be difficult to reason about when it will
+        precisely occur.  For that reason, :meth:`notify_event` should be
```

Comment:
The language here suggests that we should prefer an alternate method, but it names the same method.

Reply:
Good catch, updating the docstring.